src/sentry/workflow_engine/endpoints/organization_workflow_index.py

```diff
@@ -219,3 +235,29 @@ def post(self, request, organization):
                 bulk_validator.save()
 
         return Response(serialize(workflow, request.user), status=status.HTTP_201_CREATED)
+
+    def delete(self, request, organization):
+        """
+        Deletes workflows for a given org
+        """
+        projects = self.get_projects(request, organization)
+        if not (request.GET.get("id") or request.GET.get("query") or projects):
+            return Response(
+                {"detail": "At least one of 'id', 'query', or 'project' must be provided."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        queryset = self.filter_workflows(request, organization)
+
+        for workflow in queryset:
+            RegionScheduledDeletion.schedule(workflow, days=0, actor=request.user)
+            workflow.update(status=ObjectStatus.PENDING_DELETION)
```

Comment:
perf: instead of updating each workflow individually, you can update the entire queryset at once
```suggestion
queryset.update(status=ObjectStatus.PENDING_DELETION)
```